myjsonrpc.pm

```diff
@@ -14,6 +14,21 @@ use bmwqemu ();
 use constant DEBUG_JSON => $ENV{PERL_MYJSONRPC_DEBUG} || 0;
 use constant READ_BUFFER => $ENV{PERL_MYJSONRPC_BYTES} || 8_000_000;
 
+# hash for keeping state
+our $sockets;
+
+sub _syswrite($to_fd, $json) { syswrite($to_fd, $json) }
+
+sub is_debug () { DEBUG_JSON || $bmwqemu::vars{DEBUG_JSON_RPC} }
+
+sub wait_read ($fd) {
```

Comment:
The function is not waiting for anything:
```suggestion
sub handle_read_error ($fd) {
```
(Of course the function `can_read` is also badly named because that one *does* the waiting and is not called `wait_…`.)